py_trees/behaviours.py

```diff
@@ -736,3 +737,47 @@ def update(self) -> common.Status:
                 "|".join(["T" if result else "F" for result in results])
             )
             return common.Status.FAILURE
+
+
+class ProbabilisticBehaviour(behaviour.Behaviour):
+    """
+    Return a status based on a probability distribution.
+
+    That is, :data:`~py_trees.common.Status.SUCCESS` with probability success_probability,
+    :data:`~py_trees.common.Status.FAILURE` with probability failure_probability and
+    :data:`~py_trees.common.Status.RUNNING` with probability running_probability.
+    
+    If unspecified - a uniform distribution will be used.
+
+    Args:
+        name: name of the behaviour
+        success_probability: probability of returning :data:`~py_trees.common.Status.SUCCESS`
+        failure_probability: probability of returning :data:`~py_trees.common.Status.FAILURE`
+        running_probability: probability of returning :data:`~py_trees.common.Status.RUNNING`
+
+    .. note:: Probability distribution does not need to be normalised, it will be normalised internally.
+
+    Raises:
+        ValueError if only some of the probabilities are specified
+
+    """
+
+    def __init__(self, name: str, weights: typing.Optional[typing.List[float]] = None):
+        assert weights is None or len(weights) == 3, "either all or none of the probabilities must be specified"
```

Comment:
Prefer `ValueError` here (as indicated in the docstring).

Asserts can be optimised away and as this could result from the actions of an external user input, the library always needs to be able to reject it lest we cause silent/surprising/unhandled failures for the user.

Reply:
Done